### PR TITLE
Properly check limit in gl.activeTexture()

### LINKED ISSUE #20531: Properly check for limit in WebGLRenderingContext.activeTexture
https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.3
https://www.khronos.org/registry/OpenGL-Refpages/es2.0/xhtml/glActiveTexture.xml

> Specifies which texture unit to make active. The number of texture units is implementation dependent, but must be at least 8. texture must be one of `GL_TEXTUREi`, where `i` ranges from 0 to `(GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS - 1)`. The initial value is `GL_TEXTURE0`.

We never check `GL_MAX_COMBINED_TEXTURE_IMAGE_UNITS`, we should probably do that when getting other limits necessary for other checks, and then cache everything.

### PR DESCRIPTION
**Pull request by nox**
Fixes #20531.

This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21341)

### CODE REVIEW

File: components/canvas/webgl_thread.rs [hunk 1]
```diff
@@ -97,11 +98,20 @@ impl<VR: WebVRRenderHandler + 'static, OB: WebGLThreadObserver> WebGLThread<VR,
                                     .expect("WebGLContext not found");
                     let glsl_version = Self::get_glsl_version(ctx);
 
+                    let mut max_combined_texture_image_units = [0];
+                    unsafe {
+                        ctx.gl().get_integer_v(
```
Comment (jdm) [inline components/canvas/webgl_thread.rs]: This duplicates https://github.com/servo/rust-offscreen-rendering-context/blob/master/src/gl_limits.rs#L86.

File: components/canvas_traits/webgl.rs [hunk 2]
```diff
@@ -66,7 +66,9 @@ pub struct WebGLCreateContextResult {
     /// How the WebGLContext is shared with WebRender.
     pub share_mode: WebGLContextShareMode,
     /// The GLSL version supported by the context.
-    pub glsl_version: WebGLSLVersion
+    pub glsl_version: WebGLSLVersion,
+    /// The maximum number of combined texture image units.
+    pub max_combined_texture_image_units: u32,
```
Comment (jdm) [inline components/canvas_traits/webgl.rs]: This duplicates https://github.com/servo/rust-offscreen-rendering-context/blob/master/src/gl_limits.rs#L11.

**Review by jdm — CHANGES_REQUESTED** [review]
I haven't looked at the rest of the changes, but this stuck out to me.

File: components/canvas/webgl_thread.rs [hunk 3]
```diff
@@ -87,6 +87,7 @@ impl<VR: WebVRRenderHandler + 'static, OB: WebGLThreadObserver> WebGLThread<VR,
     }
 
     /// Handles a generic WebGLMsg message
+    #[allow(unsafe_code)]
```
Comment (avadacatavra) [inline components/canvas/webgl_thread.rs]: I don't think this is needed

**Review by avadacatavra — CHANGES_REQUESTED** [review]
lgtm

File: components/canvas/webgl_thread.rs (hunk 3 above)
Comment (nox) [reply components/canvas/webgl_thread.rs]: Good catch. I removed it.

**Review by avadacatavra — APPROVED** [review]

**Comment by jdm** [comment]
@bors-servo retry
* #21492

**Pull request merged**